Extension/src/githubAPI.ts

```diff
@@ -0,0 +1,152 @@
+}
+
+// Determines whether there exists a build that should be installed; returns the build if there is
+function getTargetBuild(builds: Build[], updateChannel: string): Build | undefined {
+    // Get predicates to determine the build to install, if any
+    let needsUpdate: (v1: PackageVersion, v2: PackageVersion) => boolean;
+    let useBuild: (build: Build) => boolean;
+    if (updateChannel === 'Insiders') {
+        needsUpdate = function(v1: PackageVersion, v2: PackageVersion): boolean { return v2.isGreaterThan(v1); };
+        useBuild = function(build: Build): boolean { return true; };
+    } else if (updateChannel === 'Default') {
+        needsUpdate = function(v1: PackageVersion, v2: PackageVersion): boolean { return v1.isGreaterThan(v2); };
+        useBuild = function(build: Build): boolean { return build.name.indexOf('-') === -1; };
+    } else {
+        throw new Error('Incorrect updateChannel setting provided');
+    }
+
+    // Get the build to install
+    const targetBuild: Build = builds.find((build) => useBuild(build));
+    if (!targetBuild) {
+        throw new Error('Failed to determine installation candidate');
+    }
+
+    // Check current version against target's version to determine if the installation should happen
+    const userVersion: PackageVersion = new PackageVersion(util.packageJson.version);
+    const targetVersion: PackageVersion = new PackageVersion(targetBuild.name);
+    if (!needsUpdate(userVersion, targetVersion)) {
+        return undefined;
+    }
+
+    return targetBuild;
+}
+
+async function getReleaseJson(): Promise<Build[]> {
+    return new Promise<Build[]>((resolve, reject) => {
+        // Create temp file to hold json
+        tmp.file(async (err, releaseJsonPath, fd, cleanupCallback) => {
+            if (err) {
+                cleanupCallback();
+                reject(new Error('Failed to create release json file'));
+            }
+
+            const releaseUrl: string = 'https://api.github.com/repos/Microsoft/vscode-cpptools/releases';
+            const header: OutgoingHttpHeaders = { 'User-Agent': 'vscode-cpptools' };
+            util.downloadFileToDestination(releaseUrl, releaseJsonPath, header)
+                .catch(() => { cleanupCallback(); reject(new Error('Failed to download release json')); })
+                // Read json
+                .then(() => { return util.readFileText(releaseJsonPath); })
+                .catch(() => { cleanupCallback(); reject(new Error('Failed to read release json file')); })
+                // Parse json
+                .then(fileContent => { cleanupCallback(); return Promise.resolve(fileContent); })
+                .then(JSON.parse)
+                .catch(() => { reject(new Error('Failed to parse release json')); })
+                // Type check and return
+                .then(releaseJson => {
+                    isArrayOfBuilds ? resolve(releaseJson) : reject('Release json is not Build[]');
```

Comment:
looks like a potential remnant on the syntax we discussed on Friday.  Make sure you pass in `releaseJson` to `isArrayOfBuilds` here.